src/sentry/remote_config/docs/protocol.md

```diff
@@ -22,17 +22,66 @@ Retrieve a project's configuration.
 
 **Feature Object**
 
-| Field | Type   | Description                        |
-| ----- | ------ | ---------------------------------- |
-| key   | string | The name used to lookup a feature. |
-| value | any    | A JSON value.                      |
+| Field    | Type           | Description                                                                     |
+| -------- | -------------- | ------------------------------------------------------------------------------- |
+| key      | string         | The name used to lookup a feature.                                              |
+| value    | any            | A JSON value.                                                                   |
+| variants | array[Variant] | Features can have any number of possible variants which alter the return value. |
+
+Features may specify zero or more variants which may alter the evaluation result.
+
+**Variant Object**
+
+| Field   | Type        | Description                                                                               |
+| ------- | ----------- | ----------------------------------------------------------------------------------------- |
+| rollout | Rollout     | An object which contains instructions for rolling out a variant.                          |
+| rules   | array[Rule] | An array of rules which all must evaluate to true in order to return the variant's value. |
+| value   | any         | The value of the feature if the variant is successfully evaluated.                        |
+
+When evaluating a set of rules and varaints each rule is a logical `AND` while each variant is a logical `OR`. Failure to evaluate one variant does not prevent another variant from successfully evaluating. Failure to evaluate one rule does invalidate the remaining rules and the evaluation proceedure can eagerly terminate.
```

Comment:
variant order is important. sdks need to respect it.

Reply:
Yes this needs to be called out explicitly in the blueprint 👍